solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/MarcosJr/MJLoginViewController.swift

```diff
@@ -31,48 +31,50 @@ class MJLoginViewController: UIViewController {
+        let windowScene = scenes.first as? UIWindowScene
+        let window = windowScene?.windows.first
+        
+        window?.rootViewController = viewController
+        window?.makeKeyAndVisible()
+    }
 
     func verifyLogin() {
         if let _ = UserDefaultsManager.UserInfos.shared.readSesion() {
             let vc = UINavigationController(rootViewController: HomeViewController())
-            let scenes = UIApplication.shared.connectedScenes
-            let windowScene = scenes.first as? UIWindowScene
-            let window = windowScene?.windows.first
-            window?.rootViewController = vc
-            window?.makeKeyAndVisible()
+            updateRootViewController(viewController: vc)
         }
     }
     
-    @IBAction func loginButton(_ sender: Any) {
+    func isConnected() -> Bool {
         if !ConnectivityManager.shared.isConnected {
             let alertController = UIAlertController(title: "Sem conexão", message: "Conecte-se à internet para tentar novamente", preferredStyle: .alert)
             let actin = UIAlertAction(title: "Ok", style: .default)
             alertController.addAction(actin)
             present(alertController, animated: true)
-            return
+            return false
+        } else {
+            return true
         }
-
-        showLoading()
+    }
+    
+    func loginRequest() {
+        let endpoint = Endpoints.Auth.login
         let parameters: [String: String] = ["email": emailTextField.text!,
                                             "password": passwordTextField.text!]
-        let endpoint = Endpoints.Auth.login
+        
+        getUserData(endpoint: endpoint, parameters: parameters)
+    }
+    
+    func getUserData(endpoint: String, parameters: [String: String]) {
         AF.request(endpoint, method: .get, parameters: parameters, headers: nil) { result in
             DispatchQueue.main.async {
                 self.stopLoading()
                 switch result {
                 case .success(let data):
-                    let decoder = JSONDecoder()
-                    if let session = try? decoder.decode(Session.self, from: data) {
-                        let vc = UINavigationController(rootViewController: HomeViewController())
-                        let scenes = UIApplication.shared.connectedScenes
-                        let windowScene = scenes.first as? UIWindowScene
-                        let window = windowScene?.windows.first
-                        window?.rootViewController = vc
-                        window?.makeKeyAndVisible()
-                        UserDefaultsManager.UserInfos.shared.save(session: session, user: nil)
-                    } else {
-                        Globals.alertMessage(title: "Ops..", message: "Houve um problema, tente novamente mais tarde.", targetVC: self)
-                    }
+                    self.saveUser(data: data)
                 case .failure:
```

Comment:
normalmente em chamadas de api, eu quebro assim:
func que chama a api
func para tratar o result
func para tratar a falha do result
func para tratar o sucesso do result
mais func se for necessário (tipo se o caso de sucesso tiver muito codigo)